pmd-java/src/main/resources/category/java/codestyle.xml

```diff
@@ -1351,6 +1351,42 @@ public class ClassInDefaultPackage {
         </example>
     </rule>
 
+    <rule name="UseUnderscoresInNumericLiterals"
+          language="java"
+          since="6.10.0"
+          minimumLanguageVersion="1.7"
+          message="Numeric literals with more than 3 digits must use '_' as a separator"
+          class="net.sourceforge.pmd.lang.rule.XPathRule"
+          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_codestyle.html#useunderscoresinnumericliterals">
+        <description>
+            Numeric literals with more than 3 digits must use '_' as a separator.
+        </description>
+        <priority>3</priority>
+        <properties>
+            <property name="version" value="2.0"/>
+            <property name="xpath">
+                <value>
+                    <![CDATA[
+//Literal[@IntLiteral = true() or
+ @LongLiteral = true() or
+ @DoubleLiteral = true() or
+ @FloatLiteral = true()]
+ [not (matches(@Image, "^(0([xb]?[0-9a-fA-F]+)?|[1-9][0-9]{0,2}(_[0-9]{3})*)(l|L|\.[0-9_]+)?([eE][\+-]?[0-9]+)?[dDfF]?$"))]
```

Comment:
I think instead of ignoring bases other than decimal, it would be nice to require them to be underscored too.
E.g. hex literals are sometimes divided in groups of two digits, bc two hex digits make a byte. 

But then, the maximum acceptable length would need to be configured base by base.

Better keep this PR simple and implement that later. I'd prefer the acceptable length property for decimal literals to be added in this PR though.

Reply:
Since, the bases other than decimal follow a different grouping strategy and 3 digit grouping does not work well with these bases, it has been ignored. We can extend this rule to support other bases later on.
I will implement the acceptable length property for decimal literals in this PR.